src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -131,6 +140,40 @@ bool HasConditionalDirectives()
 
                         return false;
                     }
+
+                    static bool IsSingleThrowNotImplementedOperation(IOperation firstBlock)
+                    {
+                        var compilation = firstBlock.SemanticModel.Compilation;
+                        var notImplementedExceptionType = compilation.NotImplementedExceptionType();
+                        if (notImplementedExceptionType == null)
+                            return false;
+
+                        if (!(firstBlock is IBlockOperation block))
+                            return false;
+
+                        var explicitOperations = block.Operations.WhereAsArray(op => !op.IsImplicit);
```

Comment:
Wouldn't this create a copy of almost every operation block array? Seems expensive for a filter check.

Reply:
That's a great point.  I'll tweak.